src/apps/omis/apps/edit/controllers/assignees.js

```
@@ -7,30 +7,46 @@ const { Order } = require('../../../models')
 
 class EditAssigneesController extends EditController {
   async configure (req, res, next) {
-    const orderId = get(res.locals, 'order.id')
-    const token = get(req.session, 'token')
-    const advisers = await getAdvisers(token)
-    const assignees = await Order.getAssignees(token, orderId)
-    const options = advisers.results.map(transformObjectToOption)
-
-    res.locals.order.assignees = assignees
-    req.form.options.fields.assignees.options = sortBy(options, 'label')
-    super.configure(req, res, next)
+    try {
+      const orderId = get(res.locals, 'order.id')
+      const canEditOrder = get(res.locals, 'order.canEditOrder')
+      const canEditAdvisers = get(res.locals, 'order.canEditAdvisers')
+      const token = get(req.session, 'token')
+      const advisers = await getAdvisers(token)
+      const assignees = await Order.getAssignees(token, orderId)
+      const options = advisers.results.map(transformObjectToOption)
+
+      req.form.options.fields.assignees.options = sortBy(options, 'label')
+      req.form.options.fields.assignees.canRemove = canEditOrder
```

could this be done using `set`?

---

I think that's a bit unnecessary and over-guarding. If the `assignees` property doesn't exist this controller should probably blow up. Don't want it to silently fail.